repository.go

```diff
@@ -1720,6 +1720,52 @@ func (r *Repository) resolveHashPrefix(hashStr string) []plumbing.Hash {
 	return hashes
 }
 
+// GetNote retuns the note for the given commit. returns a empty string, if the commit has no note.
+func (r *Repository) GetNote(commit *object.Commit) (string, error) {
+	notes, err := r.Notes()
+	if err != nil {
+		return "", err
+	}
+
+	noteRef, err := notes.Next()
+	if err != nil {
+		if errors.Is(err, io.EOF) {
+			return "", nil
+		}
+		return "", err
+	}
```

Comment:
Not sure I am missing something, but this looks like we are iterating through the first note only?

Reply:
No, this is the first ref, not the first Note.  Notes are always in the `notes/commits` ref, so we won't have more than one ref.

Reply:
Thanks for clarifying.